docs: rewrite dev containers documentation for GA

mafredri opened the pull request:
This PR rewrites the dev containers documentation for GA readiness.

The previous documentation had inaccuracies: SSH examples used the deprecated `--container` flag instead of the agent-based pattern (`coder ssh workspace.agent`), port forwarding focused on `appPort` when native forwarding to sub-agents is the primary method, and prerequisites implied users needed to "enable" dev containers when it is on by default.

Renamed admin docs folder from `devcontainers/` to `envbuilder/` since it documents Envbuilder specifically, not native dev containers. Added notes pointing to native dev containers as the recommended approach for most use cases, with Envbuilder as an alternative for Docker-less environments. Also fixes module registry URLs that pointed to `dev.registry` instead of `registry.coder.com`.

Adds a new customization guide documenting Coder-specific `devcontainer.json` options including custom agent naming, display apps configuration, custom apps with health checks, and variable interpolation using `${localEnv:...}`, Coder variables, and `$SESSION_TOKEN`.

Moves customization content from the admin docs to the user guide where users are more likely to find it. The admin docs now link to the user guide instead of duplicating content.

Improves accuracy around dev container discovery: adds the third standard devcontainer.json location for monorepos (linked to the spec), explains that discovery scans the working directory for Git repos, and clarifies that autostart behavior depends on configuration. Also fixes cross-references between docs to avoid duplication.

Corrects template descriptions: the docker-devcontainer template uses the native Dev Containers integration, while AWS and Kubernetes templates use Envbuilder. Adds example template link to the admin docs.

Refs #18907

mafredri committed:
docs: rewrite dev containers documentation for GA

The previous documentation had inaccuracies: SSH examples used the
deprecated `--container` flag instead of the agent-based pattern
(`coder ssh workspace.agent`), port forwarding focused on `appPort` when
native forwarding to sub-agents is the primary method, and prerequisites
implied users needed to "enable" dev containers when it is on by default.

Renamed admin docs folder from `devcontainers/` to `envbuilder/` since it
documents Envbuilder specifically, not native dev containers. Added notes
pointing to native dev containers as the recommended approach for most
use cases.

Also fixes module registry URLs that pointed to dev.registry instead of
registry.coder.com.

Refs #18907

mafredri committed:
docs: add dev container customization guide

Documents Coder-specific `devcontainer.json` customizations including
custom agent naming, display apps configuration, custom apps with
health checks, and variable interpolation using `${localEnv:...}`,
Coder variables, and `$SESSION_TOKEN`.

mafredri committed:
docs: move customizations from admin to user guide

Centralizes the customization documentation in the user guide where
users are more likely to find it. The admin docs now link to the
user guide instead of duplicating content.

mafredri commented:
## 📚 Documentation Check

### ✨ No Changes Needed

**Reason**: This PR is a comprehensive documentation update itself. The PR successfully rewrites dev containers documentation for GA readiness with the following comprehensive changes:

**User Guide Updates:**
- ✅ Rewrote main dev containers overview (`docs/user-guides/devcontainers/index.md`)
- ✅ Enhanced working with dev containers guide
- ✅ Expanded troubleshooting documentation
- ✅ **Added new customization guide** (`docs/user-guides/devcontainers/customizing-dev-containers.md`) documenting Coder-specific `devcontainer.json` options

**Admin Guide Updates:**
- ✅ Renamed `devcontainers/` folder to `envbuilder/` to better reflect its purpose
- ✅ Updated template extension guide (`docs/admin/templates/extending-templates/devcontainers.md`)
- ✅ Fixed module registry URLs (`dev.registry` → `registry.coder.com`)
- ✅ Moved customization content to user guide with proper cross-references

**Structure Updates:**
- ✅ Updated `manifest.json` with new documentation structure
- ✅ Added URL redirects in `docs/_redirects` for renamed paths
- ✅ Updated navigation and cross-references throughout

The documentation correctly:
- Fixes SSH examples to use agent-based pattern (`coder ssh workspace.agent` instead of deprecated `--container` flag)
- Clarifies that dev containers are enabled by default
- Positions Envbuilder as an alternative for Docker-less environments
- Provides comprehensive customization options for developers

All user-facing documentation has been thoroughly updated in this PR.

---
*This comment was generated by an AI Agent through [Coder Tasks](https://coder.com/docs/ai-coder/tasks)*

mafredri committed:
docs: improve dev containers accuracy and clarity

- Add third devcontainer.json location for monorepos, link to spec
- Clarify hostname suffix is a deployment setting
- Tighten project discovery section, clarify autostart behavior

mafredri committed:
docs: clarify discovery behavior and improve cross-references

- Explain discovery scans working directory for Git repos
- Cross-reference standard devcontainer.json locations instead of repeating
- Clarify 'How it works' steps depend on auto-start configuration
- Add link to admin docs for autostart env var
- Fix 'Currently available' phrasing to avoid staleness

mafredri committed:
fix(docs): correct devcontainer template descriptions

The docker-devcontainer template uses the Dev Containers integration,
not Envbuilder. The AWS and Kubernetes templates use Envbuilder.

Also adds example template link to the Dev Containers admin doc.

mafredri committed:
docs: s/alice/me/

mtojek commented on docs/admin/templates/extending-templates/devcontainers.md:
```diff
@@ -36,7 +38,7 @@ to ensure the `@devcontainers/cli` is installed in your workspace:
 module "devcontainers-cli" {
   count    = data.coder_workspace.me.start_count
-  source   = "dev.registry.coder.com/modules/devcontainers-cli/coder"
+  source   = "registry.coder.com/coder/devcontainers-cli/coder"
```
I might have missed this switch. Is this intended?

mafredri committed:
docs(devcontainers): note multi-repo workspace support

mtojek commented on docs/user-guides/devcontainers/working-with-dev-containers.md:
````diff
@@ -5,93 +5,159 @@ visual representation of the running environment:
+coder config-ssh
+```
+
+This creates SSH host entries for all your workspaces and their agents,
````
Perhaps add a sentence clarifying when (if anytime) the user needs to refresh config-ssh. For instance, when a new container or agent is created.

mtojek commented on docs/user-guides/devcontainers/working-with-dev-containers.md:
````diff
@@ -5,93 +5,159 @@ visual representation of the running environment:
+```
+
+This creates SSH host entries for all your workspaces and their agents,
+including dev container sub-agents. You can then connect using the standard
````
Should we describe what happens behind the scenes?

mtojek commented on docs/user-guides/devcontainers/customizing-dev-containers.md:
```diff
@@ -0,0 +1,309 @@
+| `slug`        | string  | Unique identifier for the app (required)                      |
+| `displayName` | string  | Human-readable name shown in the UI                           |
+| `url`         | string  | URL to open (supports variable interpolation)                 |
+| `command`     | string  | Command to run instead of opening a URL                       |
```
do we have an example for this property?

mafredri replied:
I'm not sure, I don't think it's worth diving deeper here as it's essentially `coder_app` settings 1-to-1.

mtojek commented on docs/user-guides/devcontainers/index.md:
```diff
@@ -1,87 +1,137 @@
+
+## Getting started
+
+### 1. Add a devcontainer.json
```
nit: instead of ### just **1. Add a devcontainer.json** (bold)?

mafredri replied:
My preference would be to drop the `1.` instead as the `**` would become load-bearing (if you remove it, it becomes an actual bullet list resulting in messed up numbering).

mtojek reviewed (commented):
I didn't check the correctness of the commands, rather focused on features and step descriptions.

mafredri committed:
fix(docs/user-guides/devcontainers): address review feedback

Remove numbered prefixes from Getting Started headers and fix config-ssh
section to correctly describe the wildcard host entry instead of multiple
entries.

mtojek reviewed (approved):
Nothing blocking, this will be a great improvement 👍

mafredri committed:
docs: s/dirty/outdated/g

mafredri committed:
docs: fix link

mafredri committed:
docs: remove useless detail

david-fraley commented:
## 📚 Documentation Check

### ✨ No Changes Needed

This PR is a comprehensive documentation rewrite that includes all necessary updates:

**What was updated:**
- ✅ Complete rewrite of dev containers user guides (`docs/user-guides/devcontainers/`)
  - New `index.md` with accurate prerequisites, features, and getting started guide
  - New `customizing-dev-containers.md` with comprehensive customization options
  - Updated `working-with-dev-containers.md` with correct SSH patterns (agent-based, not `--container` flag)
  - Expanded `troubleshooting-dev-containers.md` with more diagnostic guidance

- ✅ Renamed and clarified admin docs (`docs/admin/templates/managing-templates/envbuilder/`)
  - Folder renamed from `devcontainers/` to `envbuilder/` to accurately reflect that these docs are for Envbuilder specifically
  - Added notes distinguishing native dev containers (recommended) from Envbuilder (Docker-less alternative)
  - Fixed module registry URLs from `dev.registry` to `registry.coder.com`

- ✅ Updated cross-references throughout the docs to avoid duplication and maintain consistency

- ✅ Updated `docs/manifest.json` with new structure

- ✅ Added redirects in `docs/_redirects` for moved content

**Why no additional changes needed:**
This PR is specifically a documentation update for GA readiness. It corrects inaccuracies (deprecated SSH flags, incorrect port forwarding guidance, prerequisite confusion) and adds comprehensive new documentation for customization features. All user-facing changes are properly documented within the PR itself.

---
*This comment was generated by an AI Agent through [Coder Tasks](https://coder.com/docs/ai-coder/tasks)*

david-fraley reviewed (dismissed):
Can we include some callout that modules in our terraform registry are not currently able to be used by the sub agents? I don't want to dig and hint at all of these issues the customer is calling out, but being more clear about not being able to run agent modules would be good imo
https://codercom.slack.com/archives/C06UK9XP3F0/p1763750997864699

Good to go otherwise. I'd be ok with that as a follow up later; I'd rather see the changes get in sooner so

mafredri committed:
docs: add note about tf resources and modules
